### DISCUSSION TITLE
New Smartthings OCF Integration

### DISCUSSION THREAD

**Discussion by veista** [veista/smartthings#86] [Polls]
Would you be willing to pay for an integration that fully supports your **OCF** devices?

Due to the immense interest to this integration I would like to build a new one that runs alongside the official Smartthings integration. This current integration is not maintainable since it was only meant for minor tweaks for myself adding missing functionality which the basic Smartthings API does not provide.
I have no need for this new integration myself so I would be asking money for it and would be releasing it closed source until I have gathered 2000€ from the integration. After that I would be releasing it as open source.

The payment would grant you access to the integration + 1 issue resolution (1 unsupported device addition to the integration). Also you have the option to add your github or real name to readme.

**Comment by TheSimpleZ** [comment]
I could imagine donating to an open-source project to get prioritized issue resolution (get my device integrated). However, I have a hard time seeing myself pay for something closed-source. What if you never reach your target and suddenly don't have time to maintain the project?

**Reply by veista** [reply]
Then somebody would have to pay the remaining amount to make it open source. You can always maintain it yourself, you just would not be able to share the whole integration exept to those who have paid.

I understand your concern, reality is that this current integration is just a bubblegum patch to suit my needs, I never thought it would get this much attention. Rewriting the whole integration to better suit pure OCF is a lot of work and I have other things to do. I'm just making sure that if I spend couple of months worth of my free time I get compensated. A good software does not have issues to solve so nobody would pay.

**Comment by Koky05** [comment]
I so sorry but mine only concern to use you integration was to add WindFree feature into AC controller on top of original Integration in HA. I could live without it. I swich PowerConsumption (it is missing in your integration now and it is available in original integration) for WindFree mode. And that is not enough to pay for it. I do not use any other Samsung device and it will not change in near future.
Make it OpenSource and add  a free donation option, that is more common way. I now it is not so fast but still you can make 5€ or 2000€, no one guaranteed to you that anyone will buy it or donate you.

**Reply by veista** [reply]
I understand. I agree completely and I have the same situation. That is why I want to get compensated if I make a more maintainable Integration. Making it open source from the get-go is 90% of the work done so anybody could just pick of where I left and I would get nothing.

**Reply by Koky05** [reply]
@veista According native HA integration for SmartThings start support WindFree feature I switch back. Wish you good luck and you do great job.

**Comment by ColinSainsbury** [comment]
I was planning to donate as a thank you. I am OK with others benefiting from my donation but I also understand that you feel justified in asking for users to fund your time when requesting the addition of a new device or class of devices.

**Reply by veista** [reply]
Nice to hear! Unfortunately developing and testing the current integration is not easy and the code is pretty bad for supporting new devices.

Supporting the new integration benefits all those willing to pay and everybody once it is made open source.

**Reply by ColinSainsbury** [reply]
My assumption is that if you take funds for developing the integration for the new device, you would support it should any changes be made by either Samsung or HomeAssistant that break the integration until at least the point that the integration becomes open source. Is that a valid assumption and a commitment you are prepared to make?

**Reply by veista** [reply]
There are no guaranties in life. I cannot promise something which I do not know. Samsung may discontinue smartthings at any moment and will not accept returns on your devices, Im surely not able to fix that issue if it happens. The payment and closed sourcness just enables me to get compensated at least a little for the amount of work that would be ahead of me. The payment buys you one issue resolved that benefits everybody and is a step closer to being open source, nothing more.

**Comment by omerfaruk-aran** [comment]
Hello,

I've been following the discussion about the new integration proposal, and I wanted to share my thoughts on this matter.

Firstly, I completely understand the desire to seek compensation for your work on the new integration. However, I believe it's important to consider the current landscape and user preferences when it comes to smart home automation.

It's evident that building a user base and collecting €2000 for the integration will take a significant amount of time, especially considering that only 8 people have expressed their willingness to pay within the initial 20-day period. Many users tend to opt for free smart home automation solutions and are cautious about investing in additional modules with high price points when they are accustomed to free alternatives.

Considering the time it will take to develop the software and gather the necessary funds, it's reasonable to expect that circumstances could change during this timeframe. For instance, a competitor might release a similar integration, or Home Assistant could provide full support for the devices in question, altering the demand for your integration.

In my opinion, a more effective approach might involve collaborating with the community to expedite the development process. Releasing a basic version of the integration with essential features, without charging a fee upfront, could be enticing for users. This would not only garner more interest but also make it easier to gather the desired funds when the time comes.

By working together with the community and releasing an initial version without fees, you'll likely attract more users and generate genuine enthusiasm for your project. This strategy would also provide you with a platform to receive feedback and improve the integration based on user experiences.

Ultimately, I believe that taking a collaborative and user-focused approach will yield better results in the long run. It's important to create a sense of value and trust among potential users before transitioning to a paid model. This way, you'll have a stronger foundation to reach your fundraising goal and ensure the success of your integration.

Thank you for considering these thoughts, and I look forward to seeing how this discussion unfolds.

Best regards,

**Comment by blair287** [comment]
I'm not sure what's going on but for smartthings integrations seems so bad for samsung appliances id be willing to donate even per device to be added but seems no updates or fixes have been implemented for months anyway or have you now started with your closed source?

**Reply by veista** [reply]
I have started the project, but I have limited time at the moment.

**Reply by blair287** [reply]
Anyway to contribute and test once a contribution has been made?

**Reply by ChecksumDK** [reply]
I would like to contribute with coding or testing as well.

**Comment by llamafilm** [comment]
Why not merge all these improvements into Home Assistant core?  I generally try to avoid HACS integrations if possible.  I've just found this integration while searching online for cooktop support.  I see that you have implemented `custom.cooktopOperatingState` already which is what I needed.  I don't understand the desire to maintain a separate integration with overlapping functionality.

**Reply by veista** [reply]
I agree and many have tried. You are more than welcome to do so. It was never a desire, but merely a necessity.

**Comment by Matthieu-Em** [comment]
Hello,
I just received my new SmartThings oven, I play with this fork and it is very promising, does this donation plan is always planned? Something started?
Thanks!

**Reply by veista** [reply]
Hi, all donations are welcome. The upcoming integration is still a work in progress.

**Comment by killtux** [comment]
@veista any news of the state of the OCF integration? since the current one isn't working anymore and new samsung smartthings tokens are no more longer valid than 24h hours. is there a chance for a fix for the current integration?

**Reply by veista** [reply]
It is unfortunate that Samsung has changed this policy. This also affects the integration I have in the works. Does the official integration also have this issue?

**Reply by Koky05** [reply]
Yes, by Samsung it is recommended to change login flow from PET to OAuth method. But no integration has implemented it now.

**Comment by peaceful108** [comment]
Hi @veista , 
If you have some free time, I'm willing to pay for your time and effort to help me better adapt your Smartthings AC integration to my two Samsung Windfree AC's. 
My AC's have a few optional presets that aren't showing up in HA (like: quiet, comfort, sleep...) and I could use your assistance to make them work better with HA.
Hope you can help 🙏

**Reply by veista** [reply]
Hi. I have been meaning to work some more on this. It is good to know there is still interest.

**Reply by peaceful108** [reply]
Let me know if I can send you any info. 
And how I should proceed with regard to payment.

**Reply by veista** [reply]
All donations are welcome, you can find info on the front page. At this point in time I am not able to directly help you with your issue. I need to get a better integration up and running first.

**Comment by ColinSainsbury** [comment]
There is a new developer [https://github.com/joostlek](url) driving the SmartThings core integration forward. He is making great strides but needs help. He is running a Discord [https://discord.gg/8UqnYVzr](url) where he is looking for input

**Reply by llamafilm** [reply]
Thanks for sharing, this is great news! I see the release announcement in 2025.3. I'll join that's server and see if I can help.
BTW joostlek is not new, he's been a core home assistant maintainer for a long time.

**Comment by ColinSainsbury** [comment]
Yeah, I meant he has jumped into the SmartThings arena taking over from the previous dev.

**Comment by peaceful108** [comment]
Hi Friends, 
Those of you that are looking to add the Quiet and Fast Turbo modes to the SmartThings Custom integration, because they are not showing for your Windfree AC's, I managed to find a solution that works for my two ARTIK051_KRAC_18K AC units. 

Please see the fix in the following issue:
https://github.com/veista/smartthings/issues/131
